src/main/java/com/google/sps/servlets/LoadEventServlet.java

```diff
@@ -44,42 +44,64 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
     Key keyRequested;
     try {
       keyRequested = getEventKey(request);
-      if (keyRequested != null) {
-        Query query = new Query("Event", keyRequested);
-
-        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-        Entity eventRequested = datastore.prepare(query).asSingleEntity();
-        int alreadySaved = -1;
-
-        String userToken = request.getParameter("userToken");
-        if (userToken != null) {
-          if (Firebase.isUserLoggedIn(userToken)) {
-            String userID = Firebase.authenticateUser(userToken);
-            Key userKey = KeyFactory.createKey("User", userID);
-            List<String> tags = (List<String>) eventRequested.getProperty("tags");
-            Entity userEntity = null;
-            try {
-              userEntity = datastore.get(userKey);
-              alreadySaved = UserServlet.alreadySaved(eventRequested.getKey().getId(), userEntity);
-            } catch (EntityNotFoundException exception) {
-              userEntity = Utils.makeUserEntity(userID, false);
-              LOGGER.info("No entity found for " + userID + ", creating one now.");
-            }
-            int delta =
-                Interactions.recordInteraction(
-                    userID, keyRequested.getId(), Interactions.VIEW_SCORE, false);
-            Interactions.updatePrefs(userEntity, tags, delta);
-            datastore.put(userEntity);
-          }
-        }
-        request = populateRequest(request, eventRequested, alreadySaved);
-
-        request.getRequestDispatcher("/WEB-INF/jsp/display-event.jsp").forward(request, response);
-      }
-    } catch (IllegalArgumentException | IOException | NullPointerException | ServletException e) {
+    } catch (IllegalArgumentException | IOException e) {
       LOGGER.info("Could not retrieve event " + e);
       request.getRequestDispatcher("/WEB-INF/jsp/event-not-found.jsp").forward(request, response);
+      return;
+    }
+
+    Query query = new Query("Event", keyRequested);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity eventRequested = datastore.prepare(query).asSingleEntity();
+    int alreadySaved = -1;
+
+    String userToken = null;
+    try {
+      userToken = request.getParameter("userToken");
+      if (userToken == null) {
+        throw new IOException("No user token");
+      }
+    } catch (IOException e) {
+      request = populateRequest(request, eventRequested, alreadySaved);
+      request.getRequestDispatcher("/WEB-INF/jsp/display-event.jsp").forward(request, response);
+      return;
     }
```

Comment:
Have you considered doing: 

String userToken = request.getParameter("userToken");
if(userToken == null) {
  LOGGER.warning("No user token"); // log instead of throwing error
  request = populateRequest(request, eventRequested, alreadySaved);
  request.getRequestDispatcher("/WEB-INF/jsp/display-event.jsp").forward(request, response);
  return;
}

Reply:
Will do, thanks!